src/player.ts

```diff
@@ -43,7 +44,8 @@ export default class Player extends EventEmitter {
     return {
       faction: this.faction,
       data: this.data,
-      income: Reward.toString(Reward.merge([].concat(...this.events[Operator.Income].map(event => event.rewards))), true)
+      income: Reward.toString(Reward.merge([].concat(...this.events[Operator.Income].map(event => event.rewards))), true),
+      progress:  Object.values(finalScorings).forEach( (i) => ({ track: FinalTile[i], count: this.eventConditionCount(finalScorings[i])}))
```

Comment:
I think you mean to use `map` instead of `forEach`.

Also, `i` corresponds to a FinalTile enum, so `track` is `i`

Reply:
I was thinking to have a progress on all possible scorings, which help for advTechtile and stdTiles ( e.g. sectors, PlanetType). We could exclude what's not in the available research tiles and final scoring tiles